### ISSUE TITLE
Pantheon compiled, but does not work as expected

### ISSUE THREAD

**Issue by Azimoth** [pimvullers/elementary#157]
Hi,
I'm fairly new to Gentoo, yet I just managed to install Pantheon 

1. layman -a elementary
2. Ensure that X USE flag is included in the system's global USE flags.
3. emerge --ask pantheon-base/pantheon-shell
Let Portage take care of keywords, use flags etc.
4. emerge --ask pantheon-base/pantheon-shell
5. Done. 

Now I can select "Pantheon" as a desktop environment at the login screen, yet there's only the "dock" at the bottom, the rest of the screen is black and my curser remains a "X".
The autostarting programs do show up, but they have no "window frame", no menu, no "minimize/maximize/close".
Somewhere in between there is a message that no system tray is found.

Gnome continues to work fine. Any idea how to fix this?
What further information would you need to help me?

**Comment by pimvullers** [comment]
Hi, it seems that gala isn't started properly. Please try starting gala from a console: gala --display=.0 or something similar, or alternatively mutter (Gnome's window manager).

**Comment by Azimoth** [comment]
This is the output...

~ $ gala --display=0

(gala:6583): Clutter-CRITICAL **: Unable to initialize Clutter: Unable to open display '0'

(gala:6583): mutter-WARNING **: Unable to initialize Clutter.

**Comment by pimvullers** [comment]
You ran this after your Pantheon session is started?
Possibly the display ID is different. Please check the Xorg log to find out the correct ID

**Comment by Azimoth** [comment]
I tried this as well logging in via CLI as from Pantheon (although I cannot open a Terminal window in Pantheon, so I used ctrl+alt+F1 to get to the CLI but AFTER logging in using Pantheon).

I tried display=0 through 5 (it seemed improbable that it's a larger number, or am I wrong?) but the output did not change...

**Comment by pimvullers** [comment]
try display=:0

**Comment by Azimoth** [comment]
ok, now we're talking. It worked. Yet... how do I tell Gentoo to do this automatically without "gala --display=:0"?

**Comment by pimvullers** [comment]
System tray is wingpanel, you should be able to start this the same way as gala (if you have a terminal in the X11 session, there is no need to specify the display)

I recall that I also had this problem initially, I think I made it an autostarted application. Although now I cannot find it anymore where it gets started, so I need to dig a bit deeper to find this info.

**Comment by Azimoth** [comment]
Yes, I was able to start wingpanel from the command line (I have the terminal as an autostart-app now).
Thanks a lot for your help so far! Let me know if you remember your solution to this!

**Comment by Azimoth** [comment]
I installed Gentoo (and hence GNOME) following "Sakaki's EFI Install Guide" and there was a section on [Installing GNOME 3](https://wiki.gentoo.org/wiki/Sakaki%27s_EFI_Install_Guide/Setting_up_the_GNOME_3_Desktop#Installing_GNOME_3), where it says:

Replace existing text with the below to start GNOME
~/.xinitrc
export XDG_MENU_PREFIX=gnome-
exec gnome-session

Could this be the origin of my troubles?

**Comment by pimvullers** [comment]
in this file replace both times gnome by pantheon and give it a try.

**Comment by Azimoth** [comment]
Nope, that did not change anything...

**Comment by pimvullers** [comment]
which login manager do you use? gdm or lightdm or other or none?
I'm using lightDM,

**Comment by Azimoth** [comment]
I'm not sure to be honest... I started with a "standard" Gnome installation as described in Sakaki's Guide and later on installed the elementary overlay... I don't think the login screen changed so I assume it's the same as before, which would probably be gdm...?

**Comment by pimvullers** [comment]
and in gdm you can then select the pantheon session as an option...

Possibly you can add gala and cerbere also as autostart applications.

@Maffblaster do you recall if you had such issues, and how you resolved it?

**Comment by Azimoth** [comment]
Just edited the post above (should I delete it there and put it down here?)

**Comment by pimvullers** [comment]
ok, I had to refresh to get it. Yes this is a bug somehow slingshot should be blacklisted in lightdm.

Is it the hplip that complains about no system tray (this is what I have)? 
Because indeed wingpanel provides a panel but no system tray. It only has the wingpanel-indicators

**Comment by Azimoth** [comment]
Well... see for yourself...

This is the output of 
~ $ sudo cat /var/log/messages | grep tray
Sep 22 00:04:06 Trinity gnome-session-b[3511]: there's no system tray. Use KDE or GNOME, please!
Sep 22 00:04:06 Trinity gnome-session-b[3511]: No systemtrayicon available
Sep 22 00:04:07 Trinity gnome-session-b[3511]: No systemtrayicon available
Sep 22 00:04:09 Trinity gnome-session-b[3511]: No systemtrayicon available
Sep 22 00:05:09 Trinity gnome-session-b[3511]: error:  No system tray detected on this system.  Unable to start, exiting. 
Sep 22 00:05:09 Trinity python2.7[3597]: hp-systray(qt4)[3597]: error:  No system tray detected on this system.  Unable to start, exiting.
Sep 22 01:14:32 Trinity gnome-session-b[3471]: Starting Dropbox...there's no system tray. Use KDE or GNOME, please!
Sep 22 01:14:33 Trinity gnome-session-b[3471]: No systemtrayicon available
Sep 22 01:14:34 Trinity gnome-session-b[3471]: No systemtrayicon available
Sep 22 01:14:34 Trinity gnome-session-b[3471]: No systemtrayicon available
Sep 22 01:15:34 Trinity gnome-session-b[3471]: error:  No system tray detected on this system.  Unable to start, exiting. 
Sep 22 01:15:34 Trinity python2.7[3564]: hp-systray(qt4)[3564]: error:  No system tray detected on this system.  Unable to start, exiting.
Sep 22 01:50:55 Trinity gnome-session-b[3512]: there's no system tray. Use KDE or GNOME, please!
Sep 22 01:50:55 Trinity gnome-session-b[3512]: Starting Dropbox...No systemtrayicon available
Sep 22 01:50:56 Trinity gnome-session-b[3512]: No systemtrayicon available
Sep 22 01:50:57 Trinity gnome-session-b[3512]: No systemtrayicon available

(Trinity being the computer name...)

**Comment by pimvullers** [comment]
ok, clear. this is something that you'll have to live with for now. Or you should find out how they fixed it in elementaryOS, and then we can port/package the necessary code.

**Comment by Azimoth** [comment]
Mainly I'd like to get gala and wingpanel to start automatically...
